plugins/content/attachment/services/AttachmentAssetService.php

```diff
@@ -494,11 +494,11 @@ public function getAction($attachmentAssetId)
 	 * List attachment Assets by filter and pager
 	 * 
 	 * @action list
-	 * @param KalturaAssetFilter $filter
+	 * @param KalturaAttachmentAssetFilter $filter
 	 * @param KalturaFilterPager $pager
 	 * @return KalturaAttachmentAssetListResponse
 	 */
-	function listAction(KalturaAssetFilter $filter = null, KalturaFilterPager $pager = null)
+	function listAction(KalturaAttachmentAssetFilter $filter = null, KalturaFilterPager $pager = null)
```

Comment:
There's no compatibility today. If you pass anything other than the KalturaAttachmentFilter, and error is thrown. Line 509 attempts to cast a filter which is not of type KalturaAttachmentAssetFilter into the type. This will throw an error with any class but KalturaAttachmentAssetFilter  and KalturaTranscriptAssetFilter